game-app/game-core/src/main/java/games/strategy/triplea/attachments/RulesAttachment.java

```diff
@@ -55,6 +55,8 @@ public class RulesAttachment extends AbstractPlayerRulesAttachment {
   private int techCount = -1;
   // condition for having specific relationships
   private @Nullable List<String> relationship = null;
+  // condition for checking ai player
+  private boolean checkAIPlayers = false;
```

Comment:
What do you think of allowing to set the value to `true` or `false`, to indicate if you want to check for an ai player or a non-ai player?

That can be done by making this field a `Boolean` (capitalized), which then can default to null. But if you set it explicitly, via `true` or `false`, then it would check for that?

Reply:
(This type of thing will be hard to change later, since these objects are serializable and once you make the field a `boolean` you can't change it to `Boolean`, so this is why I'm suggesting to do it now.)

Reply:
@asvitkine Thank you again. Took you advice and allow for both true/false entries, boolean/Boolean. Also renamed the method to isAIPlayers, as you suggested before.